// @polar-sh/sdk/dist/commonjs/models/components/productcreate.js
"use strict";
/*
 * Code generated by Speakeasy (https://speakeasy.com). DO NOT EDIT.
 */
var __createBinding = (this && this.__createBinding) || (Object.create ? (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    var desc = Object.getOwnPropertyDescriptor(m, k);
    if (!desc || ("get" in desc ? !m.__esModule : desc.writable || desc.configurable)) {
      desc = { enumerable: true, get: function() { return m[k]; } };
    }
    Object.defineProperty(o, k2, desc);
}) : (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    o[k2] = m[k];
}));
var __setModuleDefault = (this && this.__setModuleDefault) || (Object.create ? (function(o, v) {
    Object.defineProperty(o, "default", { enumerable: true, value: v });
}) : function(o, v) {
    o["default"] = v;
});
var __importStar = (this && this.__importStar) || function (mod) {
    if (mod && mod.__esModule) return mod;
    var result = {};
    if (mod != null) for (var k in mod) if (k !== "default" && Object.prototype.hasOwnProperty.call(mod, k)) __createBinding(result, mod, k);
    __setModuleDefault(result, mod);
    return result;
};
Object.defineProperty(exports, "__esModule", { value: true });
exports.ProductCreate$ = exports.ProductCreate$outboundSchema = exports.ProductCreate$inboundSchema = exports.Prices$ = exports.Prices$outboundSchema = exports.Prices$inboundSchema = exports.Metadata$ = exports.Metadata$outboundSchema = exports.Metadata$inboundSchema = void 0;
exports.metadataToJSON = metadataToJSON;
exports.metadataFromJSON = metadataFromJSON;
exports.pricesToJSON = pricesToJSON;
exports.pricesFromJSON = pricesFromJSON;
exports.productCreateToJSON = productCreateToJSON;
exports.productCreateFromJSON = productCreateFromJSON;
const z = __importStar(require("zod"));
const primitives_js_1 = require("../../lib/primitives.js");
const schemas_js_1 = require("../../lib/schemas.js");
const attachedcustomfieldcreate_js_1 = require("./attachedcustomfieldcreate.js");
const productpricecustomcreate_js_1 = require("./productpricecustomcreate.js");
const productpricefixedcreate_js_1 = require("./productpricefixedcreate.js");
const productpricefreecreate_js_1 = require("./productpricefreecreate.js");
const subscriptionrecurringinterval_js_1 = require("./subscriptionrecurringinterval.js");
/** @internal */
exports.Metadata$inboundSchema = z.union([z.string(), z.number().int(), z.boolean()]);
/** @internal */
exports.Metadata$outboundSchema = z.union([z.string(), z.number().int(), z.boolean()]);
/**
 * @internal
 * @deprecated This namespace will be removed in future versions. Use schemas and types that are exported directly from this module.
 */
var Metadata$;
(function (Metadata$) {
    /** @deprecated use `Metadata$inboundSchema` instead. */
    Metadata$.inboundSchema = exports.Metadata$inboundSchema;
    /** @deprecated use `Metadata$outboundSchema` instead. */
    Metadata$.outboundSchema = exports.Metadata$outboundSchema;
})(Metadata$ || (exports.Metadata$ = Metadata$ = {}));
function metadataToJSON(metadata) {
    return JSON.stringify(exports.Metadata$outboundSchema.parse(metadata));
}
function metadataFromJSON(jsonString) {
    return (0, schemas_js_1.safeParse)(jsonString, (x) => exports.Metadata$inboundSchema.parse(JSON.parse(x)), `Failed to parse 'Metadata' from JSON`);
}
/** @internal */
exports.Prices$inboundSchema = z
    .union([
    productpricefreecreate_js_1.ProductPriceFreeCreate$inboundSchema,
    productpricefixedcreate_js_1.ProductPriceFixedCreate$inboundSchema,
    productpricecustomcreate_js_1.ProductPriceCustomCreate$inboundSchema,
]);
/** @internal */
exports.Prices$outboundSchema = z.union([
    productpricefreecreate_js_1.ProductPriceFreeCreate$outboundSchema,
    productpricefixedcreate_js_1.ProductPriceFixedCreate$outboundSchema,
    productpricecustomcreate_js_1.ProductPriceCustomCreate$outboundSchema,
]);
/**
 * @internal
 * @deprecated This namespace will be removed in future versions. Use schemas and types that are exported directly from this module.
 */
var Prices$;
(function (Prices$) {
    /** @deprecated use `Prices$inboundSchema` instead. */
    Prices$.inboundSchema = exports.Prices$inboundSchema;
    /** @deprecated use `Prices$outboundSchema` instead. */
    Prices$.outboundSchema = exports.Prices$outboundSchema;
})(Prices$ || (exports.Prices$ = Prices$ = {}));
function pricesToJSON(prices) {
    return JSON.stringify(exports.Prices$outboundSchema.parse(prices));
}
function pricesFromJSON(jsonString) {
    return (0, schemas_js_1.safeParse)(jsonString, (x) => exports.Prices$inboundSchema.parse(JSON.parse(x)), `Failed to parse 'Prices' from JSON`);
}
/** @internal */
exports.ProductCreate$inboundSchema = z.object({
    metadata: z.record(z.union([z.string(), z.number().int(), z.boolean()]))
        .optional(),
    name: z.string(),
    description: z.nullable(z.string()).optional(),
    recurring_interval: z.nullable(subscriptionrecurringinterval_js_1.SubscriptionRecurringInterval$inboundSchema),
    prices: z.array(z.union([
        productpricefreecreate_js_1.ProductPriceFreeCreate$inboundSchema,
        productpricefixedcreate_js_1.ProductPriceFixedCreate$inboundSchema,
        productpricecustomcreate_js_1.ProductPriceCustomCreate$inboundSchema,
    ])),
    medias: z.nullable(z.array(z.string())).optional(),
    attached_custom_fields: z.array(attachedcustomfieldcreate_js_1.AttachedCustomFieldCreate$inboundSchema)
        .optional(),
    organization_id: z.nullable(z.string()).optional(),
}).transform((v) => {
    return (0, primitives_js_1.remap)(v, {
        "recurring_interval": "recurringInterval",
        "attached_custom_fields": "attachedCustomFields",
        "organization_id": "organizationId",
    });
});
/** @internal */
exports.ProductCreate$outboundSchema = z.object({
    metadata: z.record(z.union([z.string(), z.number().int(), z.boolean()]))
        .optional(),
    name: z.string(),
    description: z.nullable(z.string()).optional(),
    recurringInterval: z.nullable(subscriptionrecurringinterval_js_1.SubscriptionRecurringInterval$outboundSchema),
    prices: z.array(z.union([
        productpricefreecreate_js_1.ProductPriceFreeCreate$outboundSchema,
        productpricefixedcreate_js_1.ProductPriceFixedCreate$outboundSchema,
        productpricecustomcreate_js_1.ProductPriceCustomCreate$outboundSchema,
    ])),
    medias: z.nullable(z.array(z.string())).optional(),
    attachedCustomFields: z.array(attachedcustomfieldcreate_js_1.AttachedCustomFieldCreate$outboundSchema)
        .optional(),
    organizationId: z.nullable(z.string()).optional(),
}).transform((v) => {
    return (0, primitives_js_1.remap)(v, {
        recurringInterval: "recurring_interval",
        attachedCustomFields: "attached_custom_fields",
        organizationId: "organization_id",
    });
});
/**
 * @internal
 * @deprecated This namespace will be removed in future versions. Use schemas and types that are exported directly from this module.
 */
var ProductCreate$;
(function (ProductCreate$) {
    /** @deprecated use `ProductCreate$inboundSchema` instead. */
    ProductCreate$.inboundSchema = exports.ProductCreate$inboundSchema;
    /** @deprecated use `ProductCreate$outboundSchema` instead. */
    ProductCreate$.outboundSchema = exports.ProductCreate$outboundSchema;
})(ProductCreate$ || (exports.ProductCreate$ = ProductCreate$ = {}));
function productCreateToJSON(productCreate) {
    return JSON.stringify(exports.ProductCreate$outboundSchema.parse(productCreate));
}
function productCreateFromJSON(jsonString) {
    return (0, schemas_js_1.safeParse)(jsonString, (x) => exports.ProductCreate$inboundSchema.parse(JSON.parse(x)), `Failed to parse 'ProductCreate' from JSON`);
}
