// @polar-sh/sdk/dist/commonjs/models/components/subscription.js
"use strict";
/*
 * Code generated by Speakeasy (https://speakeasy.com). DO NOT EDIT.
 */
var __createBinding = (this && this.__createBinding) || (Object.create ? (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    var desc = Object.getOwnPropertyDescriptor(m, k);
    if (!desc || ("get" in desc ? !m.__esModule : desc.writable || desc.configurable)) {
      desc = { enumerable: true, get: function() { return m[k]; } };
    }
    Object.defineProperty(o, k2, desc);
}) : (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    o[k2] = m[k];
}));
var __setModuleDefault = (this && this.__setModuleDefault) || (Object.create ? (function(o, v) {
    Object.defineProperty(o, "default", { enumerable: true, value: v });
}) : function(o, v) {
    o["default"] = v;
});
var __importStar = (this && this.__importStar) || function (mod) {
    if (mod && mod.__esModule) return mod;
    var result = {};
    if (mod != null) for (var k in mod) if (k !== "default" && Object.prototype.hasOwnProperty.call(mod, k)) __createBinding(result, mod, k);
    __setModuleDefault(result, mod);
    return result;
};
Object.defineProperty(exports, "__esModule", { value: true });
exports.Subscription$ = exports.Subscription$outboundSchema = exports.Subscription$inboundSchema = exports.SubscriptionDiscount$ = exports.SubscriptionDiscount$outboundSchema = exports.SubscriptionDiscount$inboundSchema = exports.SubscriptionPrice$ = exports.SubscriptionPrice$outboundSchema = exports.SubscriptionPrice$inboundSchema = exports.SubscriptionCustomFieldData$ = exports.SubscriptionCustomFieldData$outboundSchema = exports.SubscriptionCustomFieldData$inboundSchema = exports.SubscriptionMetadata$ = exports.SubscriptionMetadata$outboundSchema = exports.SubscriptionMetadata$inboundSchema = void 0;
exports.subscriptionMetadataToJSON = subscriptionMetadataToJSON;
exports.subscriptionMetadataFromJSON = subscriptionMetadataFromJSON;
exports.subscriptionCustomFieldDataToJSON = subscriptionCustomFieldDataToJSON;
exports.subscriptionCustomFieldDataFromJSON = subscriptionCustomFieldDataFromJSON;
exports.subscriptionPriceToJSON = subscriptionPriceToJSON;
exports.subscriptionPriceFromJSON = subscriptionPriceFromJSON;
exports.subscriptionDiscountToJSON = subscriptionDiscountToJSON;
exports.subscriptionDiscountFromJSON = subscriptionDiscountFromJSON;
exports.subscriptionToJSON = subscriptionToJSON;
exports.subscriptionFromJSON = subscriptionFromJSON;
const z = __importStar(require("zod"));
const primitives_js_1 = require("../../lib/primitives.js");
const schemas_js_1 = require("../../lib/schemas.js");
const customercancellationreason_js_1 = require("./customercancellationreason.js");
const discountfixedonceforeverdurationbase_js_1 = require("./discountfixedonceforeverdurationbase.js");
const discountfixedrepeatdurationbase_js_1 = require("./discountfixedrepeatdurationbase.js");
const discountpercentageonceforeverdurationbase_js_1 = require("./discountpercentageonceforeverdurationbase.js");
const discountpercentagerepeatdurationbase_js_1 = require("./discountpercentagerepeatdurationbase.js");
const legacyrecurringproductprice_js_1 = require("./legacyrecurringproductprice.js");
const product_js_1 = require("./product.js");
const productprice_js_1 = require("./productprice.js");
const subscriptioncustomer_js_1 = require("./subscriptioncustomer.js");
const subscriptionrecurringinterval_js_1 = require("./subscriptionrecurringinterval.js");
const subscriptionstatus_js_1 = require("./subscriptionstatus.js");
const subscriptionuser_js_1 = require("./subscriptionuser.js");
/** @internal */
exports.SubscriptionMetadata$inboundSchema = z.union([z.string(), z.number().int(), z.boolean()]);
/** @internal */
exports.SubscriptionMetadata$outboundSchema = z.union([z.string(), z.number().int(), z.boolean()]);
/**
 * @internal
 * @deprecated This namespace will be removed in future versions. Use schemas and types that are exported directly from this module.
 */
var SubscriptionMetadata$;
(function (SubscriptionMetadata$) {
    /** @deprecated use `SubscriptionMetadata$inboundSchema` instead. */
    SubscriptionMetadata$.inboundSchema = exports.SubscriptionMetadata$inboundSchema;
    /** @deprecated use `SubscriptionMetadata$outboundSchema` instead. */
    SubscriptionMetadata$.outboundSchema = exports.SubscriptionMetadata$outboundSchema;
})(SubscriptionMetadata$ || (exports.SubscriptionMetadata$ = SubscriptionMetadata$ = {}));
function subscriptionMetadataToJSON(subscriptionMetadata) {
    return JSON.stringify(exports.SubscriptionMetadata$outboundSchema.parse(subscriptionMetadata));
}
function subscriptionMetadataFromJSON(jsonString) {
    return (0, schemas_js_1.safeParse)(jsonString, (x) => exports.SubscriptionMetadata$inboundSchema.parse(JSON.parse(x)), `Failed to parse 'SubscriptionMetadata' from JSON`);
}
/** @internal */
exports.SubscriptionCustomFieldData$inboundSchema = z.union([
    z.string(),
    z.number().int(),
    z.boolean(),
    z.string().datetime({ offset: true }).transform(v => new Date(v)),
]);
/** @internal */
exports.SubscriptionCustomFieldData$outboundSchema = z.union([
    z.string(),
    z.number().int(),
    z.boolean(),
    z.date().transform(v => v.toISOString()),
]);
/**
 * @internal
 * @deprecated This namespace will be removed in future versions. Use schemas and types that are exported directly from this module.
 */
var SubscriptionCustomFieldData$;
(function (SubscriptionCustomFieldData$) {
    /** @deprecated use `SubscriptionCustomFieldData$inboundSchema` instead. */
    SubscriptionCustomFieldData$.inboundSchema = exports.SubscriptionCustomFieldData$inboundSchema;
    /** @deprecated use `SubscriptionCustomFieldData$outboundSchema` instead. */
    SubscriptionCustomFieldData$.outboundSchema = exports.SubscriptionCustomFieldData$outboundSchema;
})(SubscriptionCustomFieldData$ || (exports.SubscriptionCustomFieldData$ = SubscriptionCustomFieldData$ = {}));
function subscriptionCustomFieldDataToJSON(subscriptionCustomFieldData) {
    return JSON.stringify(exports.SubscriptionCustomFieldData$outboundSchema.parse(subscriptionCustomFieldData));
}
function subscriptionCustomFieldDataFromJSON(jsonString) {
    return (0, schemas_js_1.safeParse)(jsonString, (x) => exports.SubscriptionCustomFieldData$inboundSchema.parse(JSON.parse(x)), `Failed to parse 'SubscriptionCustomFieldData' from JSON`);
}
/** @internal */
exports.SubscriptionPrice$inboundSchema = z.union([
    legacyrecurringproductprice_js_1.LegacyRecurringProductPrice$inboundSchema,
    productprice_js_1.ProductPrice$inboundSchema,
]);
/** @internal */
exports.SubscriptionPrice$outboundSchema = z.union([
    legacyrecurringproductprice_js_1.LegacyRecurringProductPrice$outboundSchema,
    productprice_js_1.ProductPrice$outboundSchema,
]);
/**
 * @internal
 * @deprecated This namespace will be removed in future versions. Use schemas and types that are exported directly from this module.
 */
var SubscriptionPrice$;
(function (SubscriptionPrice$) {
    /** @deprecated use `SubscriptionPrice$inboundSchema` instead. */
    SubscriptionPrice$.inboundSchema = exports.SubscriptionPrice$inboundSchema;
    /** @deprecated use `SubscriptionPrice$outboundSchema` instead. */
    SubscriptionPrice$.outboundSchema = exports.SubscriptionPrice$outboundSchema;
})(SubscriptionPrice$ || (exports.SubscriptionPrice$ = SubscriptionPrice$ = {}));
function subscriptionPriceToJSON(subscriptionPrice) {
    return JSON.stringify(exports.SubscriptionPrice$outboundSchema.parse(subscriptionPrice));
}
function subscriptionPriceFromJSON(jsonString) {
    return (0, schemas_js_1.safeParse)(jsonString, (x) => exports.SubscriptionPrice$inboundSchema.parse(JSON.parse(x)), `Failed to parse 'SubscriptionPrice' from JSON`);
}
/** @internal */
exports.SubscriptionDiscount$inboundSchema = z.union([
    discountpercentageonceforeverdurationbase_js_1.DiscountPercentageOnceForeverDurationBase$inboundSchema,
    discountfixedonceforeverdurationbase_js_1.DiscountFixedOnceForeverDurationBase$inboundSchema,
    discountpercentagerepeatdurationbase_js_1.DiscountPercentageRepeatDurationBase$inboundSchema,
    discountfixedrepeatdurationbase_js_1.DiscountFixedRepeatDurationBase$inboundSchema,
]);
/** @internal */
exports.SubscriptionDiscount$outboundSchema = z.union([
    discountpercentageonceforeverdurationbase_js_1.DiscountPercentageOnceForeverDurationBase$outboundSchema,
    discountfixedonceforeverdurationbase_js_1.DiscountFixedOnceForeverDurationBase$outboundSchema,
    discountpercentagerepeatdurationbase_js_1.DiscountPercentageRepeatDurationBase$outboundSchema,
    discountfixedrepeatdurationbase_js_1.DiscountFixedRepeatDurationBase$outboundSchema,
]);
/**
 * @internal
 * @deprecated This namespace will be removed in future versions. Use schemas and types that are exported directly from this module.
 */
var SubscriptionDiscount$;
(function (SubscriptionDiscount$) {
    /** @deprecated use `SubscriptionDiscount$inboundSchema` instead. */
    SubscriptionDiscount$.inboundSchema = exports.SubscriptionDiscount$inboundSchema;
    /** @deprecated use `SubscriptionDiscount$outboundSchema` instead. */
    SubscriptionDiscount$.outboundSchema = exports.SubscriptionDiscount$outboundSchema;
})(SubscriptionDiscount$ || (exports.SubscriptionDiscount$ = SubscriptionDiscount$ = {}));
function subscriptionDiscountToJSON(subscriptionDiscount) {
    return JSON.stringify(exports.SubscriptionDiscount$outboundSchema.parse(subscriptionDiscount));
}
function subscriptionDiscountFromJSON(jsonString) {
    return (0, schemas_js_1.safeParse)(jsonString, (x) => exports.SubscriptionDiscount$inboundSchema.parse(JSON.parse(x)), `Failed to parse 'SubscriptionDiscount' from JSON`);
}
/** @internal */
exports.Subscription$inboundSchema = z.object({
    created_at: z.string().datetime({ offset: true }).transform(v => new Date(v)),
    modified_at: z.nullable(z.string().datetime({ offset: true }).transform(v => new Date(v))),
    id: z.string(),
    amount: z.nullable(z.number().int()),
    currency: z.nullable(z.string()),
    recurring_interval: subscriptionrecurringinterval_js_1.SubscriptionRecurringInterval$inboundSchema,
    status: subscriptionstatus_js_1.SubscriptionStatus$inboundSchema,
    current_period_start: z.string().datetime({ offset: true }).transform(v => new Date(v)),
    current_period_end: z.nullable(z.string().datetime({ offset: true }).transform(v => new Date(v))),
    cancel_at_period_end: z.boolean(),
    canceled_at: z.nullable(z.string().datetime({ offset: true }).transform(v => new Date(v))),
    started_at: z.nullable(z.string().datetime({ offset: true }).transform(v => new Date(v))),
    ends_at: z.nullable(z.string().datetime({ offset: true }).transform(v => new Date(v))),
    ended_at: z.nullable(z.string().datetime({ offset: true }).transform(v => new Date(v))),
    customer_id: z.string(),
    product_id: z.string(),
    price_id: z.string(),
    discount_id: z.nullable(z.string()),
    checkout_id: z.nullable(z.string()),
    customer_cancellation_reason: z.nullable(customercancellationreason_js_1.CustomerCancellationReason$inboundSchema),
    customer_cancellation_comment: z.nullable(z.string()),
    metadata: z.record(z.union([z.string(), z.number().int(), z.boolean()])),
    custom_field_data: z.record(z.nullable(z.union([
        z.string(),
        z.number().int(),
        z.boolean(),
        z.string().datetime({ offset: true }).transform(v => new Date(v)),
    ]))).optional(),
    customer: subscriptioncustomer_js_1.SubscriptionCustomer$inboundSchema,
    user_id: z.string(),
    user: subscriptionuser_js_1.SubscriptionUser$inboundSchema,
    product: product_js_1.Product$inboundSchema,
    price: z.union([
        legacyrecurringproductprice_js_1.LegacyRecurringProductPrice$inboundSchema,
        productprice_js_1.ProductPrice$inboundSchema,
    ]),
    discount: z.nullable(z.union([
        discountpercentageonceforeverdurationbase_js_1.DiscountPercentageOnceForeverDurationBase$inboundSchema,
        discountfixedonceforeverdurationbase_js_1.DiscountFixedOnceForeverDurationBase$inboundSchema,
        discountpercentagerepeatdurationbase_js_1.DiscountPercentageRepeatDurationBase$inboundSchema,
        discountfixedrepeatdurationbase_js_1.DiscountFixedRepeatDurationBase$inboundSchema,
    ])),
}).transform((v) => {
    return (0, primitives_js_1.remap)(v, {
        "created_at": "createdAt",
        "modified_at": "modifiedAt",
        "recurring_interval": "recurringInterval",
        "current_period_start": "currentPeriodStart",
        "current_period_end": "currentPeriodEnd",
        "cancel_at_period_end": "cancelAtPeriodEnd",
        "canceled_at": "canceledAt",
        "started_at": "startedAt",
        "ends_at": "endsAt",
        "ended_at": "endedAt",
        "customer_id": "customerId",
        "product_id": "productId",
        "price_id": "priceId",
        "discount_id": "discountId",
        "checkout_id": "checkoutId",
        "customer_cancellation_reason": "customerCancellationReason",
        "customer_cancellation_comment": "customerCancellationComment",
        "custom_field_data": "customFieldData",
        "user_id": "userId",
    });
});
/** @internal */
exports.Subscription$outboundSchema = z.object({
    createdAt: z.date().transform(v => v.toISOString()),
    modifiedAt: z.nullable(z.date().transform(v => v.toISOString())),
    id: z.string(),
    amount: z.nullable(z.number().int()),
    currency: z.nullable(z.string()),
    recurringInterval: subscriptionrecurringinterval_js_1.SubscriptionRecurringInterval$outboundSchema,
    status: subscriptionstatus_js_1.SubscriptionStatus$outboundSchema,
    currentPeriodStart: z.date().transform(v => v.toISOString()),
    currentPeriodEnd: z.nullable(z.date().transform(v => v.toISOString())),
    cancelAtPeriodEnd: z.boolean(),
    canceledAt: z.nullable(z.date().transform(v => v.toISOString())),
    startedAt: z.nullable(z.date().transform(v => v.toISOString())),
    endsAt: z.nullable(z.date().transform(v => v.toISOString())),
    endedAt: z.nullable(z.date().transform(v => v.toISOString())),
    customerId: z.string(),
    productId: z.string(),
    priceId: z.string(),
    discountId: z.nullable(z.string()),
    checkoutId: z.nullable(z.string()),
    customerCancellationReason: z.nullable(customercancellationreason_js_1.CustomerCancellationReason$outboundSchema),
    customerCancellationComment: z.nullable(z.string()),
    metadata: z.record(z.union([z.string(), z.number().int(), z.boolean()])),
    customFieldData: z.record(z.nullable(z.union([
        z.string(),
        z.number().int(),
        z.boolean(),
        z.date().transform(v => v.toISOString()),
    ]))).optional(),
    customer: subscriptioncustomer_js_1.SubscriptionCustomer$outboundSchema,
    userId: z.string(),
    user: subscriptionuser_js_1.SubscriptionUser$outboundSchema,
    product: product_js_1.Product$outboundSchema,
    price: z.union([
        legacyrecurringproductprice_js_1.LegacyRecurringProductPrice$outboundSchema,
        productprice_js_1.ProductPrice$outboundSchema,
    ]),
    discount: z.nullable(z.union([
        discountpercentageonceforeverdurationbase_js_1.DiscountPercentageOnceForeverDurationBase$outboundSchema,
        discountfixedonceforeverdurationbase_js_1.DiscountFixedOnceForeverDurationBase$outboundSchema,
        discountpercentagerepeatdurationbase_js_1.DiscountPercentageRepeatDurationBase$outboundSchema,
        discountfixedrepeatdurationbase_js_1.DiscountFixedRepeatDurationBase$outboundSchema,
    ])),
}).transform((v) => {
    return (0, primitives_js_1.remap)(v, {
        createdAt: "created_at",
        modifiedAt: "modified_at",
        recurringInterval: "recurring_interval",
        currentPeriodStart: "current_period_start",
        currentPeriodEnd: "current_period_end",
        cancelAtPeriodEnd: "cancel_at_period_end",
        canceledAt: "canceled_at",
        startedAt: "started_at",
        endsAt: "ends_at",
        endedAt: "ended_at",
        customerId: "customer_id",
        productId: "product_id",
        priceId: "price_id",
        discountId: "discount_id",
        checkoutId: "checkout_id",
        customerCancellationReason: "customer_cancellation_reason",
        customerCancellationComment: "customer_cancellation_comment",
        customFieldData: "custom_field_data",
        userId: "user_id",
    });
});
/**
 * @internal
 * @deprecated This namespace will be removed in future versions. Use schemas and types that are exported directly from this module.
 */
var Subscription$;
(function (Subscription$) {
    /** @deprecated use `Subscription$inboundSchema` instead. */
    Subscription$.inboundSchema = exports.Subscription$inboundSchema;
    /** @deprecated use `Subscription$outboundSchema` instead. */
    Subscription$.outboundSchema = exports.Subscription$outboundSchema;
})(Subscription$ || (exports.Subscription$ = Subscription$ = {}));
function subscriptionToJSON(subscription) {
    return JSON.stringify(exports.Subscription$outboundSchema.parse(subscription));
}
function subscriptionFromJSON(jsonString) {
    return (0, schemas_js_1.safeParse)(jsonString, (x) => exports.Subscription$inboundSchema.parse(JSON.parse(x)), `Failed to parse 'Subscription' from JSON`);
}
