// @polar-sh/nextjs/dist/index.js
// src/checkout/checkout.ts
import { Polar } from "@polar-sh/sdk";
import { NextResponse } from "next/server";
var Checkout = ({
  accessToken,
  successUrl,
  returnUrl,
  server,
  theme,
  includeCheckoutId = true
}) => {
  const polar = new Polar({
    accessToken,
    server
  });
  return async (req) => {
    const url = new URL(req.url);
    const products = url.searchParams.getAll("products");
    if (products.length === 0) {
      return NextResponse.json(
        { error: "Missing products in query params" },
        { status: 400 }
      );
    }
    const success = successUrl ? new URL(successUrl) : void 0;
    if (success && includeCheckoutId) {
      success.searchParams.set("checkoutId", "{CHECKOUT_ID}");
    }
    const retUrl = returnUrl ? new URL(returnUrl) : void 0;
    try {
      const result = await polar.checkouts.create({
        products,
        successUrl: success ? decodeURI(success.toString()) : void 0,
        customerId: url.searchParams.get("customerId") ?? void 0,
        externalCustomerId: url.searchParams.get("customerExternalId") ?? void 0,
        customerEmail: url.searchParams.get("customerEmail") ?? void 0,
        customerName: url.searchParams.get("customerName") ?? void 0,
        customerBillingAddress: url.searchParams.has("customerBillingAddress") ? JSON.parse(url.searchParams.get("customerBillingAddress") ?? "{}") : void 0,
        customerTaxId: url.searchParams.get("customerTaxId") ?? void 0,
        customerIpAddress: url.searchParams.get("customerIpAddress") ?? void 0,
        customerMetadata: url.searchParams.has("customerMetadata") ? JSON.parse(url.searchParams.get("customerMetadata") ?? "{}") : void 0,
        allowDiscountCodes: url.searchParams.has("allowDiscountCodes") ? url.searchParams.get("allowDiscountCodes") === "true" : void 0,
        discountId: url.searchParams.get("discountId") ?? void 0,
        metadata: url.searchParams.has("metadata") ? JSON.parse(url.searchParams.get("metadata") ?? "{}") : void 0,
        returnUrl: retUrl ? decodeURI(retUrl.toString()) : void 0
      });
      const redirectUrl = new URL(result.url);
      if (theme) {
        redirectUrl.searchParams.set("theme", theme);
      }
      return NextResponse.redirect(redirectUrl.toString());
    } catch (error) {
      console.error(error);
      return NextResponse.error();
    }
  };
};

// src/customerPortal/customerPortal.ts
import { Polar as Polar2 } from "@polar-sh/sdk";
import { NextResponse as NextResponse2 } from "next/server";
var CustomerPortal = ({
  accessToken,
  server,
  getCustomerId,
  returnUrl
}) => {
  const polar = new Polar2({
    accessToken,
    server
  });
  return async (req) => {
    const retUrl = returnUrl ? new URL(returnUrl) : void 0;
    const customerId = await getCustomerId(req);
    if (!customerId) {
      return NextResponse2.json(
        { error: "customerId not defined" },
        { status: 400 }
      );
    }
    try {
      const result = await polar.customerSessions.create({
        customerId,
        returnUrl: retUrl ? decodeURI(retUrl.toString()) : void 0
      });
      return NextResponse2.redirect(result.customerPortalUrl);
    } catch (error) {
      console.error(error);
      return NextResponse2.error();
    }
  };
};

// src/webhooks/webhooks.ts
import {
  handleWebhookPayload
} from "@polar-sh/adapter-utils";
import {
  WebhookVerificationError,
  validateEvent
} from "@polar-sh/sdk/webhooks";
import { NextResponse as NextResponse3 } from "next/server";
import {
  EntitlementStrategy,
  Entitlements
} from "@polar-sh/adapter-utils";
var Webhooks = ({
  webhookSecret,
  entitlements,
  onPayload,
  ...eventHandlers
}) => {
  return async (request) => {
    const requestBody = await request.text();
    const webhookHeaders = {
      "webhook-id": request.headers.get("webhook-id") ?? "",
      "webhook-timestamp": request.headers.get("webhook-timestamp") ?? "",
      "webhook-signature": request.headers.get("webhook-signature") ?? ""
    };
    let webhookPayload;
    try {
      webhookPayload = validateEvent(
        requestBody,
        webhookHeaders,
        webhookSecret
      );
    } catch (error) {
      if (error instanceof WebhookVerificationError) {
        return NextResponse3.json({ received: false }, { status: 403 });
      }
      throw error;
    }
    await handleWebhookPayload(webhookPayload, {
      webhookSecret,
      entitlements,
      onPayload,
      ...eventHandlers
    });
    return NextResponse3.json({ received: true });
  };
};
export {
  Checkout,
  CustomerPortal,
  EntitlementStrategy,
  Entitlements,
  Webhooks
};
