// @polar-sh/nextjs/dist/index.cjs
"use strict";
var __defProp = Object.defineProperty;
var __getOwnPropDesc = Object.getOwnPropertyDescriptor;
var __getOwnPropNames = Object.getOwnPropertyNames;
var __hasOwnProp = Object.prototype.hasOwnProperty;
var __export = (target, all) => {
  for (var name in all)
    __defProp(target, name, { get: all[name], enumerable: true });
};
var __copyProps = (to, from, except, desc) => {
  if (from && typeof from === "object" || typeof from === "function") {
    for (let key of __getOwnPropNames(from))
      if (!__hasOwnProp.call(to, key) && key !== except)
        __defProp(to, key, { get: () => from[key], enumerable: !(desc = __getOwnPropDesc(from, key)) || desc.enumerable });
  }
  return to;
};
var __toCommonJS = (mod) => __copyProps(__defProp({}, "__esModule", { value: true }), mod);

// src/index.ts
var index_exports = {};
__export(index_exports, {
  Checkout: () => Checkout,
  CustomerPortal: () => CustomerPortal,
  EntitlementStrategy: () => import_adapter_utils2.EntitlementStrategy,
  Entitlements: () => import_adapter_utils2.Entitlements,
  Webhooks: () => Webhooks
});
module.exports = __toCommonJS(index_exports);

// src/checkout/checkout.ts
var import_sdk = require("@polar-sh/sdk");
var import_server = require("next/server");
var Checkout = ({
  accessToken,
  successUrl,
  returnUrl,
  server,
  theme,
  includeCheckoutId = true
}) => {
  const polar = new import_sdk.Polar({
    accessToken,
    server
  });
  return async (req) => {
    const url = new URL(req.url);
    const products = url.searchParams.getAll("products");
    if (products.length === 0) {
      return import_server.NextResponse.json(
        { error: "Missing products in query params" },
        { status: 400 }
      );
    }
    const success = successUrl ? new URL(successUrl) : void 0;
    if (success && includeCheckoutId) {
      success.searchParams.set("checkoutId", "{CHECKOUT_ID}");
    }
    const retUrl = returnUrl ? new URL(returnUrl) : void 0;
    try {
      const result = await polar.checkouts.create({
        products,
        successUrl: success ? decodeURI(success.toString()) : void 0,
        customerId: url.searchParams.get("customerId") ?? void 0,
        externalCustomerId: url.searchParams.get("customerExternalId") ?? void 0,
        customerEmail: url.searchParams.get("customerEmail") ?? void 0,
        customerName: url.searchParams.get("customerName") ?? void 0,
        customerBillingAddress: url.searchParams.has("customerBillingAddress") ? JSON.parse(url.searchParams.get("customerBillingAddress") ?? "{}") : void 0,
        customerTaxId: url.searchParams.get("customerTaxId") ?? void 0,
        customerIpAddress: url.searchParams.get("customerIpAddress") ?? void 0,
        customerMetadata: url.searchParams.has("customerMetadata") ? JSON.parse(url.searchParams.get("customerMetadata") ?? "{}") : void 0,
        allowDiscountCodes: url.searchParams.has("allowDiscountCodes") ? url.searchParams.get("allowDiscountCodes") === "true" : void 0,
        discountId: url.searchParams.get("discountId") ?? void 0,
        metadata: url.searchParams.has("metadata") ? JSON.parse(url.searchParams.get("metadata") ?? "{}") : void 0,
        returnUrl: retUrl ? decodeURI(retUrl.toString()) : void 0
      });
      const redirectUrl = new URL(result.url);
      if (theme) {
        redirectUrl.searchParams.set("theme", theme);
      }
      return import_server.NextResponse.redirect(redirectUrl.toString());
    } catch (error) {
      console.error(error);
      return import_server.NextResponse.error();
    }
  };
};

// src/customerPortal/customerPortal.ts
var import_sdk2 = require("@polar-sh/sdk");
var import_server2 = require("next/server");
var CustomerPortal = ({
  accessToken,
  server,
  getCustomerId,
  returnUrl
}) => {
  const polar = new import_sdk2.Polar({
    accessToken,
    server
  });
  return async (req) => {
    const retUrl = returnUrl ? new URL(returnUrl) : void 0;
    const customerId = await getCustomerId(req);
    if (!customerId) {
      return import_server2.NextResponse.json(
        { error: "customerId not defined" },
        { status: 400 }
      );
    }
    try {
      const result = await polar.customerSessions.create({
        customerId,
        returnUrl: retUrl ? decodeURI(retUrl.toString()) : void 0
      });
      return import_server2.NextResponse.redirect(result.customerPortalUrl);
    } catch (error) {
      console.error(error);
      return import_server2.NextResponse.error();
    }
  };
};

// src/webhooks/webhooks.ts
var import_adapter_utils = require("@polar-sh/adapter-utils");
var import_webhooks = require("@polar-sh/sdk/webhooks");
var import_server3 = require("next/server");
var import_adapter_utils2 = require("@polar-sh/adapter-utils");
var Webhooks = ({
  webhookSecret,
  entitlements,
  onPayload,
  ...eventHandlers
}) => {
  return async (request) => {
    const requestBody = await request.text();
    const webhookHeaders = {
      "webhook-id": request.headers.get("webhook-id") ?? "",
      "webhook-timestamp": request.headers.get("webhook-timestamp") ?? "",
      "webhook-signature": request.headers.get("webhook-signature") ?? ""
    };
    let webhookPayload;
    try {
      webhookPayload = (0, import_webhooks.validateEvent)(
        requestBody,
        webhookHeaders,
        webhookSecret
      );
    } catch (error) {
      if (error instanceof import_webhooks.WebhookVerificationError) {
        return import_server3.NextResponse.json({ received: false }, { status: 403 });
      }
      throw error;
    }
    await (0, import_adapter_utils.handleWebhookPayload)(webhookPayload, {
      webhookSecret,
      entitlements,
      onPayload,
      ...eventHandlers
    });
    return import_server3.NextResponse.json({ received: true });
  };
};
// Annotate the CommonJS export names for ESM import in node:
0 && (module.exports = {
  Checkout,
  CustomerPortal,
  EntitlementStrategy,
  Entitlements,
  Webhooks
});
