// @polar-sh/sdk/dist/esm/models/components/eventcreatecustomer.js
/*
 * Code generated by Speakeasy (https://speakeasy.com). DO NOT EDIT.
 */
import * as z from "zod";
import { remap as remap$ } from "../../lib/primitives.js";
import { safeParse } from "../../lib/schemas.js";
import { EventMetadataInput$inboundSchema, EventMetadataInput$outboundSchema, } from "./eventmetadatainput.js";
/** @internal */
export const EventCreateCustomer$inboundSchema = z.object({
    timestamp: z.string().datetime({ offset: true }).transform(v => new Date(v))
        .optional(),
    name: z.string(),
    organization_id: z.nullable(z.string()).optional(),
    metadata: z.record(EventMetadataInput$inboundSchema).optional(),
    customer_id: z.string(),
}).transform((v) => {
    return remap$(v, {
        "organization_id": "organizationId",
        "customer_id": "customerId",
    });
});
/** @internal */
export const EventCreateCustomer$outboundSchema = z.object({
    timestamp: z.date().transform(v => v.toISOString()).optional(),
    name: z.string(),
    organizationId: z.nullable(z.string()).optional(),
    metadata: z.record(EventMetadataInput$outboundSchema).optional(),
    customerId: z.string(),
}).transform((v) => {
    return remap$(v, {
        organizationId: "organization_id",
        customerId: "customer_id",
    });
});
/**
 * @internal
 * @deprecated This namespace will be removed in future versions. Use schemas and types that are exported directly from this module.
 */
export var EventCreateCustomer$;
(function (EventCreateCustomer$) {
    /** @deprecated use `EventCreateCustomer$inboundSchema` instead. */
    EventCreateCustomer$.inboundSchema = EventCreateCustomer$inboundSchema;
    /** @deprecated use `EventCreateCustomer$outboundSchema` instead. */
    EventCreateCustomer$.outboundSchema = EventCreateCustomer$outboundSchema;
})(EventCreateCustomer$ || (EventCreateCustomer$ = {}));
export function eventCreateCustomerToJSON(eventCreateCustomer) {
    return JSON.stringify(EventCreateCustomer$outboundSchema.parse(eventCreateCustomer));
}
export function eventCreateCustomerFromJSON(jsonString) {
    return safeParse(jsonString, (x) => EventCreateCustomer$inboundSchema.parse(JSON.parse(x)), `Failed to parse 'EventCreateCustomer' from JSON`);
}
