// @polar-sh/sdk/dist/esm/models/components/subscriptioncustomer.js
/*
 * Code generated by Speakeasy (https://speakeasy.com). DO NOT EDIT.
 */
import * as z from "zod";
import { remap as remap$ } from "../../lib/primitives.js";
import { safeParse } from "../../lib/schemas.js";
import { Address$inboundSchema, Address$outboundSchema, } from "./address.js";
import { TaxIDFormat$inboundSchema, TaxIDFormat$outboundSchema, } from "./taxidformat.js";
/** @internal */
export const SubscriptionCustomerMetadata$inboundSchema = z.union([z.string(), z.number().int(), z.boolean()]);
/** @internal */
export const SubscriptionCustomerMetadata$outboundSchema = z.union([z.string(), z.number().int(), z.boolean()]);
/**
 * @internal
 * @deprecated This namespace will be removed in future versions. Use schemas and types that are exported directly from this module.
 */
export var SubscriptionCustomerMetadata$;
(function (SubscriptionCustomerMetadata$) {
    /** @deprecated use `SubscriptionCustomerMetadata$inboundSchema` instead. */
    SubscriptionCustomerMetadata$.inboundSchema = SubscriptionCustomerMetadata$inboundSchema;
    /** @deprecated use `SubscriptionCustomerMetadata$outboundSchema` instead. */
    SubscriptionCustomerMetadata$.outboundSchema = SubscriptionCustomerMetadata$outboundSchema;
})(SubscriptionCustomerMetadata$ || (SubscriptionCustomerMetadata$ = {}));
export function subscriptionCustomerMetadataToJSON(subscriptionCustomerMetadata) {
    return JSON.stringify(SubscriptionCustomerMetadata$outboundSchema.parse(subscriptionCustomerMetadata));
}
export function subscriptionCustomerMetadataFromJSON(jsonString) {
    return safeParse(jsonString, (x) => SubscriptionCustomerMetadata$inboundSchema.parse(JSON.parse(x)), `Failed to parse 'SubscriptionCustomerMetadata' from JSON`);
}
/** @internal */
export const SubscriptionCustomerTaxId$inboundSchema = z.union([z.string(), TaxIDFormat$inboundSchema]);
/** @internal */
export const SubscriptionCustomerTaxId$outboundSchema = z.union([z.string(), TaxIDFormat$outboundSchema]);
/**
 * @internal
 * @deprecated This namespace will be removed in future versions. Use schemas and types that are exported directly from this module.
 */
export var SubscriptionCustomerTaxId$;
(function (SubscriptionCustomerTaxId$) {
    /** @deprecated use `SubscriptionCustomerTaxId$inboundSchema` instead. */
    SubscriptionCustomerTaxId$.inboundSchema = SubscriptionCustomerTaxId$inboundSchema;
    /** @deprecated use `SubscriptionCustomerTaxId$outboundSchema` instead. */
    SubscriptionCustomerTaxId$.outboundSchema = SubscriptionCustomerTaxId$outboundSchema;
})(SubscriptionCustomerTaxId$ || (SubscriptionCustomerTaxId$ = {}));
export function subscriptionCustomerTaxIdToJSON(subscriptionCustomerTaxId) {
    return JSON.stringify(SubscriptionCustomerTaxId$outboundSchema.parse(subscriptionCustomerTaxId));
}
export function subscriptionCustomerTaxIdFromJSON(jsonString) {
    return safeParse(jsonString, (x) => SubscriptionCustomerTaxId$inboundSchema.parse(JSON.parse(x)), `Failed to parse 'SubscriptionCustomerTaxId' from JSON`);
}
/** @internal */
export const SubscriptionCustomer$inboundSchema = z.object({
    id: z.string(),
    created_at: z.string().datetime({ offset: true }).transform(v => new Date(v)),
    modified_at: z.nullable(z.string().datetime({ offset: true }).transform(v => new Date(v))),
    metadata: z.record(z.union([z.string(), z.number().int(), z.boolean()])),
    external_id: z.nullable(z.string()),
    email: z.string(),
    email_verified: z.boolean(),
    name: z.nullable(z.string()),
    billing_address: z.nullable(Address$inboundSchema),
    tax_id: z.nullable(z.array(z.nullable(z.union([z.string(), TaxIDFormat$inboundSchema])))),
    organization_id: z.string(),
    avatar_url: z.string(),
}).transform((v) => {
    return remap$(v, {
        "created_at": "createdAt",
        "modified_at": "modifiedAt",
        "external_id": "externalId",
        "email_verified": "emailVerified",
        "billing_address": "billingAddress",
        "tax_id": "taxId",
        "organization_id": "organizationId",
        "avatar_url": "avatarUrl",
    });
});
/** @internal */
export const SubscriptionCustomer$outboundSchema = z.object({
    id: z.string(),
    createdAt: z.date().transform(v => v.toISOString()),
    modifiedAt: z.nullable(z.date().transform(v => v.toISOString())),
    metadata: z.record(z.union([z.string(), z.number().int(), z.boolean()])),
    externalId: z.nullable(z.string()),
    email: z.string(),
    emailVerified: z.boolean(),
    name: z.nullable(z.string()),
    billingAddress: z.nullable(Address$outboundSchema),
    taxId: z.nullable(z.array(z.nullable(z.union([z.string(), TaxIDFormat$outboundSchema])))),
    organizationId: z.string(),
    avatarUrl: z.string(),
}).transform((v) => {
    return remap$(v, {
        createdAt: "created_at",
        modifiedAt: "modified_at",
        externalId: "external_id",
        emailVerified: "email_verified",
        billingAddress: "billing_address",
        taxId: "tax_id",
        organizationId: "organization_id",
        avatarUrl: "avatar_url",
    });
});
/**
 * @internal
 * @deprecated This namespace will be removed in future versions. Use schemas and types that are exported directly from this module.
 */
export var SubscriptionCustomer$;
(function (SubscriptionCustomer$) {
    /** @deprecated use `SubscriptionCustomer$inboundSchema` instead. */
    SubscriptionCustomer$.inboundSchema = SubscriptionCustomer$inboundSchema;
    /** @deprecated use `SubscriptionCustomer$outboundSchema` instead. */
    SubscriptionCustomer$.outboundSchema = SubscriptionCustomer$outboundSchema;
})(SubscriptionCustomer$ || (SubscriptionCustomer$ = {}));
export function subscriptionCustomerToJSON(subscriptionCustomer) {
    return JSON.stringify(SubscriptionCustomer$outboundSchema.parse(subscriptionCustomer));
}
export function subscriptionCustomerFromJSON(jsonString) {
    return safeParse(jsonString, (x) => SubscriptionCustomer$inboundSchema.parse(JSON.parse(x)), `Failed to parse 'SubscriptionCustomer' from JSON`);
}
