// @polar-sh/sdk/dist/esm/models/components/subscription.js
/*
 * Code generated by Speakeasy (https://speakeasy.com). DO NOT EDIT.
 */
import * as z from "zod";
import { remap as remap$ } from "../../lib/primitives.js";
import { safeParse } from "../../lib/schemas.js";
import { CustomerCancellationReason$inboundSchema, CustomerCancellationReason$outboundSchema, } from "./customercancellationreason.js";
import { DiscountFixedOnceForeverDurationBase$inboundSchema, DiscountFixedOnceForeverDurationBase$outboundSchema, } from "./discountfixedonceforeverdurationbase.js";
import { DiscountFixedRepeatDurationBase$inboundSchema, DiscountFixedRepeatDurationBase$outboundSchema, } from "./discountfixedrepeatdurationbase.js";
import { DiscountPercentageOnceForeverDurationBase$inboundSchema, DiscountPercentageOnceForeverDurationBase$outboundSchema, } from "./discountpercentageonceforeverdurationbase.js";
import { DiscountPercentageRepeatDurationBase$inboundSchema, DiscountPercentageRepeatDurationBase$outboundSchema, } from "./discountpercentagerepeatdurationbase.js";
import { LegacyRecurringProductPrice$inboundSchema, LegacyRecurringProductPrice$outboundSchema, } from "./legacyrecurringproductprice.js";
import { Product$inboundSchema, Product$outboundSchema, } from "./product.js";
import { ProductPrice$inboundSchema, ProductPrice$outboundSchema, } from "./productprice.js";
import { SubscriptionCustomer$inboundSchema, SubscriptionCustomer$outboundSchema, } from "./subscriptioncustomer.js";
import { SubscriptionRecurringInterval$inboundSchema, SubscriptionRecurringInterval$outboundSchema, } from "./subscriptionrecurringinterval.js";
import { SubscriptionStatus$inboundSchema, SubscriptionStatus$outboundSchema, } from "./subscriptionstatus.js";
import { SubscriptionUser$inboundSchema, SubscriptionUser$outboundSchema, } from "./subscriptionuser.js";
/** @internal */
export const SubscriptionMetadata$inboundSchema = z.union([z.string(), z.number().int(), z.boolean()]);
/** @internal */
export const SubscriptionMetadata$outboundSchema = z.union([z.string(), z.number().int(), z.boolean()]);
/**
 * @internal
 * @deprecated This namespace will be removed in future versions. Use schemas and types that are exported directly from this module.
 */
export var SubscriptionMetadata$;
(function (SubscriptionMetadata$) {
    /** @deprecated use `SubscriptionMetadata$inboundSchema` instead. */
    SubscriptionMetadata$.inboundSchema = SubscriptionMetadata$inboundSchema;
    /** @deprecated use `SubscriptionMetadata$outboundSchema` instead. */
    SubscriptionMetadata$.outboundSchema = SubscriptionMetadata$outboundSchema;
})(SubscriptionMetadata$ || (SubscriptionMetadata$ = {}));
export function subscriptionMetadataToJSON(subscriptionMetadata) {
    return JSON.stringify(SubscriptionMetadata$outboundSchema.parse(subscriptionMetadata));
}
export function subscriptionMetadataFromJSON(jsonString) {
    return safeParse(jsonString, (x) => SubscriptionMetadata$inboundSchema.parse(JSON.parse(x)), `Failed to parse 'SubscriptionMetadata' from JSON`);
}
/** @internal */
export const SubscriptionCustomFieldData$inboundSchema = z.union([
    z.string(),
    z.number().int(),
    z.boolean(),
    z.string().datetime({ offset: true }).transform(v => new Date(v)),
]);
/** @internal */
export const SubscriptionCustomFieldData$outboundSchema = z.union([
    z.string(),
    z.number().int(),
    z.boolean(),
    z.date().transform(v => v.toISOString()),
]);
/**
 * @internal
 * @deprecated This namespace will be removed in future versions. Use schemas and types that are exported directly from this module.
 */
export var SubscriptionCustomFieldData$;
(function (SubscriptionCustomFieldData$) {
    /** @deprecated use `SubscriptionCustomFieldData$inboundSchema` instead. */
    SubscriptionCustomFieldData$.inboundSchema = SubscriptionCustomFieldData$inboundSchema;
    /** @deprecated use `SubscriptionCustomFieldData$outboundSchema` instead. */
    SubscriptionCustomFieldData$.outboundSchema = SubscriptionCustomFieldData$outboundSchema;
})(SubscriptionCustomFieldData$ || (SubscriptionCustomFieldData$ = {}));
export function subscriptionCustomFieldDataToJSON(subscriptionCustomFieldData) {
    return JSON.stringify(SubscriptionCustomFieldData$outboundSchema.parse(subscriptionCustomFieldData));
}
export function subscriptionCustomFieldDataFromJSON(jsonString) {
    return safeParse(jsonString, (x) => SubscriptionCustomFieldData$inboundSchema.parse(JSON.parse(x)), `Failed to parse 'SubscriptionCustomFieldData' from JSON`);
}
/** @internal */
export const SubscriptionPrice$inboundSchema = z.union([
    LegacyRecurringProductPrice$inboundSchema,
    ProductPrice$inboundSchema,
]);
/** @internal */
export const SubscriptionPrice$outboundSchema = z.union([
    LegacyRecurringProductPrice$outboundSchema,
    ProductPrice$outboundSchema,
]);
/**
 * @internal
 * @deprecated This namespace will be removed in future versions. Use schemas and types that are exported directly from this module.
 */
export var SubscriptionPrice$;
(function (SubscriptionPrice$) {
    /** @deprecated use `SubscriptionPrice$inboundSchema` instead. */
    SubscriptionPrice$.inboundSchema = SubscriptionPrice$inboundSchema;
    /** @deprecated use `SubscriptionPrice$outboundSchema` instead. */
    SubscriptionPrice$.outboundSchema = SubscriptionPrice$outboundSchema;
})(SubscriptionPrice$ || (SubscriptionPrice$ = {}));
export function subscriptionPriceToJSON(subscriptionPrice) {
    return JSON.stringify(SubscriptionPrice$outboundSchema.parse(subscriptionPrice));
}
export function subscriptionPriceFromJSON(jsonString) {
    return safeParse(jsonString, (x) => SubscriptionPrice$inboundSchema.parse(JSON.parse(x)), `Failed to parse 'SubscriptionPrice' from JSON`);
}
/** @internal */
export const SubscriptionDiscount$inboundSchema = z.union([
    DiscountPercentageOnceForeverDurationBase$inboundSchema,
    DiscountFixedOnceForeverDurationBase$inboundSchema,
    DiscountPercentageRepeatDurationBase$inboundSchema,
    DiscountFixedRepeatDurationBase$inboundSchema,
]);
/** @internal */
export const SubscriptionDiscount$outboundSchema = z.union([
    DiscountPercentageOnceForeverDurationBase$outboundSchema,
    DiscountFixedOnceForeverDurationBase$outboundSchema,
    DiscountPercentageRepeatDurationBase$outboundSchema,
    DiscountFixedRepeatDurationBase$outboundSchema,
]);
/**
 * @internal
 * @deprecated This namespace will be removed in future versions. Use schemas and types that are exported directly from this module.
 */
export var SubscriptionDiscount$;
(function (SubscriptionDiscount$) {
    /** @deprecated use `SubscriptionDiscount$inboundSchema` instead. */
    SubscriptionDiscount$.inboundSchema = SubscriptionDiscount$inboundSchema;
    /** @deprecated use `SubscriptionDiscount$outboundSchema` instead. */
    SubscriptionDiscount$.outboundSchema = SubscriptionDiscount$outboundSchema;
})(SubscriptionDiscount$ || (SubscriptionDiscount$ = {}));
export function subscriptionDiscountToJSON(subscriptionDiscount) {
    return JSON.stringify(SubscriptionDiscount$outboundSchema.parse(subscriptionDiscount));
}
export function subscriptionDiscountFromJSON(jsonString) {
    return safeParse(jsonString, (x) => SubscriptionDiscount$inboundSchema.parse(JSON.parse(x)), `Failed to parse 'SubscriptionDiscount' from JSON`);
}
/** @internal */
export const Subscription$inboundSchema = z.object({
    created_at: z.string().datetime({ offset: true }).transform(v => new Date(v)),
    modified_at: z.nullable(z.string().datetime({ offset: true }).transform(v => new Date(v))),
    id: z.string(),
    amount: z.nullable(z.number().int()),
    currency: z.nullable(z.string()),
    recurring_interval: SubscriptionRecurringInterval$inboundSchema,
    status: SubscriptionStatus$inboundSchema,
    current_period_start: z.string().datetime({ offset: true }).transform(v => new Date(v)),
    current_period_end: z.nullable(z.string().datetime({ offset: true }).transform(v => new Date(v))),
    cancel_at_period_end: z.boolean(),
    canceled_at: z.nullable(z.string().datetime({ offset: true }).transform(v => new Date(v))),
    started_at: z.nullable(z.string().datetime({ offset: true }).transform(v => new Date(v))),
    ends_at: z.nullable(z.string().datetime({ offset: true }).transform(v => new Date(v))),
    ended_at: z.nullable(z.string().datetime({ offset: true }).transform(v => new Date(v))),
    customer_id: z.string(),
    product_id: z.string(),
    price_id: z.string(),
    discount_id: z.nullable(z.string()),
    checkout_id: z.nullable(z.string()),
    customer_cancellation_reason: z.nullable(CustomerCancellationReason$inboundSchema),
    customer_cancellation_comment: z.nullable(z.string()),
    metadata: z.record(z.union([z.string(), z.number().int(), z.boolean()])),
    custom_field_data: z.record(z.nullable(z.union([
        z.string(),
        z.number().int(),
        z.boolean(),
        z.string().datetime({ offset: true }).transform(v => new Date(v)),
    ]))).optional(),
    customer: SubscriptionCustomer$inboundSchema,
    user_id: z.string(),
    user: SubscriptionUser$inboundSchema,
    product: Product$inboundSchema,
    price: z.union([
        LegacyRecurringProductPrice$inboundSchema,
        ProductPrice$inboundSchema,
    ]),
    discount: z.nullable(z.union([
        DiscountPercentageOnceForeverDurationBase$inboundSchema,
        DiscountFixedOnceForeverDurationBase$inboundSchema,
        DiscountPercentageRepeatDurationBase$inboundSchema,
        DiscountFixedRepeatDurationBase$inboundSchema,
    ])),
}).transform((v) => {
    return remap$(v, {
        "created_at": "createdAt",
        "modified_at": "modifiedAt",
        "recurring_interval": "recurringInterval",
        "current_period_start": "currentPeriodStart",
        "current_period_end": "currentPeriodEnd",
        "cancel_at_period_end": "cancelAtPeriodEnd",
        "canceled_at": "canceledAt",
        "started_at": "startedAt",
        "ends_at": "endsAt",
        "ended_at": "endedAt",
        "customer_id": "customerId",
        "product_id": "productId",
        "price_id": "priceId",
        "discount_id": "discountId",
        "checkout_id": "checkoutId",
        "customer_cancellation_reason": "customerCancellationReason",
        "customer_cancellation_comment": "customerCancellationComment",
        "custom_field_data": "customFieldData",
        "user_id": "userId",
    });
});
/** @internal */
export const Subscription$outboundSchema = z.object({
    createdAt: z.date().transform(v => v.toISOString()),
    modifiedAt: z.nullable(z.date().transform(v => v.toISOString())),
    id: z.string(),
    amount: z.nullable(z.number().int()),
    currency: z.nullable(z.string()),
    recurringInterval: SubscriptionRecurringInterval$outboundSchema,
    status: SubscriptionStatus$outboundSchema,
    currentPeriodStart: z.date().transform(v => v.toISOString()),
    currentPeriodEnd: z.nullable(z.date().transform(v => v.toISOString())),
    cancelAtPeriodEnd: z.boolean(),
    canceledAt: z.nullable(z.date().transform(v => v.toISOString())),
    startedAt: z.nullable(z.date().transform(v => v.toISOString())),
    endsAt: z.nullable(z.date().transform(v => v.toISOString())),
    endedAt: z.nullable(z.date().transform(v => v.toISOString())),
    customerId: z.string(),
    productId: z.string(),
    priceId: z.string(),
    discountId: z.nullable(z.string()),
    checkoutId: z.nullable(z.string()),
    customerCancellationReason: z.nullable(CustomerCancellationReason$outboundSchema),
    customerCancellationComment: z.nullable(z.string()),
    metadata: z.record(z.union([z.string(), z.number().int(), z.boolean()])),
    customFieldData: z.record(z.nullable(z.union([
        z.string(),
        z.number().int(),
        z.boolean(),
        z.date().transform(v => v.toISOString()),
    ]))).optional(),
    customer: SubscriptionCustomer$outboundSchema,
    userId: z.string(),
    user: SubscriptionUser$outboundSchema,
    product: Product$outboundSchema,
    price: z.union([
        LegacyRecurringProductPrice$outboundSchema,
        ProductPrice$outboundSchema,
    ]),
    discount: z.nullable(z.union([
        DiscountPercentageOnceForeverDurationBase$outboundSchema,
        DiscountFixedOnceForeverDurationBase$outboundSchema,
        DiscountPercentageRepeatDurationBase$outboundSchema,
        DiscountFixedRepeatDurationBase$outboundSchema,
    ])),
}).transform((v) => {
    return remap$(v, {
        createdAt: "created_at",
        modifiedAt: "modified_at",
        recurringInterval: "recurring_interval",
        currentPeriodStart: "current_period_start",
        currentPeriodEnd: "current_period_end",
        cancelAtPeriodEnd: "cancel_at_period_end",
        canceledAt: "canceled_at",
        startedAt: "started_at",
        endsAt: "ends_at",
        endedAt: "ended_at",
        customerId: "customer_id",
        productId: "product_id",
        priceId: "price_id",
        discountId: "discount_id",
        checkoutId: "checkout_id",
        customerCancellationReason: "customer_cancellation_reason",
        customerCancellationComment: "customer_cancellation_comment",
        customFieldData: "custom_field_data",
        userId: "user_id",
    });
});
/**
 * @internal
 * @deprecated This namespace will be removed in future versions. Use schemas and types that are exported directly from this module.
 */
export var Subscription$;
(function (Subscription$) {
    /** @deprecated use `Subscription$inboundSchema` instead. */
    Subscription$.inboundSchema = Subscription$inboundSchema;
    /** @deprecated use `Subscription$outboundSchema` instead. */
    Subscription$.outboundSchema = Subscription$outboundSchema;
})(Subscription$ || (Subscription$ = {}));
export function subscriptionToJSON(subscription) {
    return JSON.stringify(Subscription$outboundSchema.parse(subscription));
}
export function subscriptionFromJSON(jsonString) {
    return safeParse(jsonString, (x) => Subscription$inboundSchema.parse(JSON.parse(x)), `Failed to parse 'Subscription' from JSON`);
}
