// @polar-sh/sdk/dist/commonjs/mcp-server/tools/customersDelete.js
"use strict";
/*
 * Code generated by Speakeasy (https://speakeasy.com). DO NOT EDIT.
 */
Object.defineProperty(exports, "__esModule", { value: true });
exports.tool$customersDelete = void 0;
const customersDelete_js_1 = require("../../funcs/customersDelete.js");
const customersdelete_js_1 = require("../../models/operations/customersdelete.js");
const tools_js_1 = require("../tools.js");
const args = {
    request: customersdelete_js_1.CustomersDeleteRequest$inboundSchema,
};
exports.tool$customersDelete = {
    name: "customers_delete",
    description: `Delete Customer

Delete a customer.

Immediately cancels any active subscriptions and revokes any active benefits.

**Scopes**: \`customers:write\``,
    scopes: ["write", "customers"],
    args,
    tool: async (client, args, ctx) => {
        const [result, apiCall] = await (0, customersDelete_js_1.customersDelete)(client, args.request, { fetchOptions: { signal: ctx.signal } }).$inspect();
        if (!result.ok) {
            return {
                content: [{ type: "text", text: result.error.message }],
                isError: true,
            };
        }
        return (0, tools_js_1.formatResult)(void 0, apiCall);
    },
};
