// @polar-sh/sdk/dist/esm/models/components/eventcreateexternalcustomer.js
/*
 * Code generated by Speakeasy (https://speakeasy.com). DO NOT EDIT.
 */
import * as z from "zod";
import { remap as remap$ } from "../../lib/primitives.js";
import { safeParse } from "../../lib/schemas.js";
import { EventMetadataInput$inboundSchema, EventMetadataInput$outboundSchema, } from "./eventmetadatainput.js";
/** @internal */
export const EventCreateExternalCustomer$inboundSchema = z.object({
    timestamp: z.string().datetime({ offset: true }).transform(v => new Date(v))
        .optional(),
    name: z.string(),
    organization_id: z.nullable(z.string()).optional(),
    metadata: z.record(EventMetadataInput$inboundSchema).optional(),
    external_customer_id: z.string(),
}).transform((v) => {
    return remap$(v, {
        "organization_id": "organizationId",
        "external_customer_id": "externalCustomerId",
    });
});
/** @internal */
export const EventCreateExternalCustomer$outboundSchema = z.object({
    timestamp: z.date().transform(v => v.toISOString()).optional(),
    name: z.string(),
    organizationId: z.nullable(z.string()).optional(),
    metadata: z.record(EventMetadataInput$outboundSchema).optional(),
    externalCustomerId: z.string(),
}).transform((v) => {
    return remap$(v, {
        organizationId: "organization_id",
        externalCustomerId: "external_customer_id",
    });
});
/**
 * @internal
 * @deprecated This namespace will be removed in future versions. Use schemas and types that are exported directly from this module.
 */
export var EventCreateExternalCustomer$;
(function (EventCreateExternalCustomer$) {
    /** @deprecated use `EventCreateExternalCustomer$inboundSchema` instead. */
    EventCreateExternalCustomer$.inboundSchema = EventCreateExternalCustomer$inboundSchema;
    /** @deprecated use `EventCreateExternalCustomer$outboundSchema` instead. */
    EventCreateExternalCustomer$.outboundSchema = EventCreateExternalCustomer$outboundSchema;
})(EventCreateExternalCustomer$ || (EventCreateExternalCustomer$ = {}));
export function eventCreateExternalCustomerToJSON(eventCreateExternalCustomer) {
    return JSON.stringify(EventCreateExternalCustomer$outboundSchema.parse(eventCreateExternalCustomer));
}
export function eventCreateExternalCustomerFromJSON(jsonString) {
    return safeParse(jsonString, (x) => EventCreateExternalCustomer$inboundSchema.parse(JSON.parse(x)), `Failed to parse 'EventCreateExternalCustomer' from JSON`);
}
