// @polar-sh/sdk/dist/esm/models/components/userevent.js
/*
 * Code generated by Speakeasy (https://speakeasy.com). DO NOT EDIT.
 */
import * as z from "zod";
import { remap as remap$ } from "../../lib/primitives.js";
import { safeParse } from "../../lib/schemas.js";
import { Customer$inboundSchema, Customer$outboundSchema, } from "./customer.js";
import { EventMetadataOutput$inboundSchema, EventMetadataOutput$outboundSchema, } from "./eventmetadataoutput.js";
/** @internal */
export const UserEvent$inboundSchema = z.object({
    id: z.string(),
    timestamp: z.string().datetime({ offset: true }).transform(v => new Date(v)),
    organization_id: z.string(),
    customer_id: z.nullable(z.string()),
    customer: z.nullable(Customer$inboundSchema),
    external_customer_id: z.nullable(z.string()),
    name: z.string(),
    source: z.literal("user"),
    metadata: z.record(EventMetadataOutput$inboundSchema),
}).transform((v) => {
    return remap$(v, {
        "organization_id": "organizationId",
        "customer_id": "customerId",
        "external_customer_id": "externalCustomerId",
    });
});
/** @internal */
export const UserEvent$outboundSchema = z.object({
    id: z.string(),
    timestamp: z.date().transform(v => v.toISOString()),
    organizationId: z.string(),
    customerId: z.nullable(z.string()),
    customer: z.nullable(Customer$outboundSchema),
    externalCustomerId: z.nullable(z.string()),
    name: z.string(),
    source: z.literal("user"),
    metadata: z.record(EventMetadataOutput$outboundSchema),
}).transform((v) => {
    return remap$(v, {
        organizationId: "organization_id",
        customerId: "customer_id",
        externalCustomerId: "external_customer_id",
    });
});
/**
 * @internal
 * @deprecated This namespace will be removed in future versions. Use schemas and types that are exported directly from this module.
 */
export var UserEvent$;
(function (UserEvent$) {
    /** @deprecated use `UserEvent$inboundSchema` instead. */
    UserEvent$.inboundSchema = UserEvent$inboundSchema;
    /** @deprecated use `UserEvent$outboundSchema` instead. */
    UserEvent$.outboundSchema = UserEvent$outboundSchema;
})(UserEvent$ || (UserEvent$ = {}));
export function userEventToJSON(userEvent) {
    return JSON.stringify(UserEvent$outboundSchema.parse(userEvent));
}
export function userEventFromJSON(jsonString) {
    return safeParse(jsonString, (x) => UserEvent$inboundSchema.parse(JSON.parse(x)), `Failed to parse 'UserEvent' from JSON`);
}
